### DISCUSSION TITLE
Tilling by Way of Orchard?

### DISCUSSION THREAD

**Discussion by stefanesa** [polymathrobotics/caladan_examples#18] [Q&A]
Hey all,

I was working on my tillage behavior earlier and, midway through tilling the field, clicked to tell the tractor to go to the far side of the orchard.  The tractor kept driving to it's next tillage goal, then drove to the far side of the orchard as commanded...and then drove back around to the tillage goal after that.

I tried replicating the issue and instead the tractor just ignored my set command (see below)

<img width="653" alt="image" src="https://user-images.githubusercontent.com/1851242/180889118-b8b45076-bd14-44a9-89f5-3a115c50b9a0.png">

Is that normal?  Does a new command not interrupt the old commands?

**Comment by iliabaranov** [comment]
Our navigation system has a queue size of 1.
Hence, if the robot is already heading to a goal, you can send a new goal. However, it will not go to the new goal until it has arrived at the current goal. If you want to immediately go to a new goal, first issue a stop command, and then send a new goal.
